contrib/epee/include/net/abstract_http_client.h

```diff
@@ -70,6 +71,7 @@ namespace http
     virtual bool connect(std::chrono::milliseconds timeout) = 0;
     virtual bool disconnect() = 0;
     virtual bool is_connected(bool *ssl = NULL) = 0;
+    virtual std::error_code last_error();
```

Comment:
@woodser a new method was added to the abtract http interface. The default behavior is to return a generic `not_connected` code when not connected, but can be extended to return more useful messages.

Reply:
What would be returned when connected and there is no error?

Could the error not be thrown from the other methods for standard exception handling?

Reply:
> What would be returned when connected and there is no error?

A default constructed empty `error_code` indicating no failure.

> Could the error not be thrown from the other methods for standard exception handling?

From where exactly? The `connect` function? The helper `invoke` functions?

The `connect` method uses `bool` for status and catches all exceptions, the control flow for every call site would have to be changed and inspected. `invoke` uses bool too, but doesn't try to catch exceptions.